src/coffee-script.coffee

```diff
@@ -62,6 +62,7 @@ exports.compile = compile = withPrettyErrors (code, options) ->
   js = ""
   for fragment in fragments
     # Update the sourcemap with data from each fragment
+    fragment.annotateCode()
```

Comment:
This is definitely not the way to do this. Your annotation comment would be another type of token, like block comments. You need to define them in grammar.coffee.